securing_apps/topics/oidc/java/servlet-filter-adapter.adoc

```diff
@@ -64,3 +64,91 @@ To use this filter, include this maven artifact in your WAR poms:
+
+If you are using Apache Karaf, you can simply install a feature from the Keycloak feature repo:
+
+[source,subs="attributes+"]
+----
+karaf@root()> feature:repo-add mvn:org.keycloak/keycloak-osgi-features/{project_versionMvn}/xml/features
+karaf@root()> feature:install keycloak-servlet-filter-adapter
+----
+
+For other OSGi runtimes, please refer to the runtime documentation on how to install the adapter bundle and its dependencies.
+
+====== Configuration
+
+First, the adapter needs to be registered as an OSGi component. The easiest way to do that is to extend `org.keycloak.adapters.servlet.KeycloakOIDCFilter` and to decorate it with OSGi annotations:
+
+[source,java]
+----
+package mypackage;
+
+import javax.servlet.Filter;
+import org.keycloak.adapters.servlet.KeycloakOIDCFilter;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.http.whiteboard.HttpWhiteboardConstants;
+
+@Component(
+    immediate = true,
+    service = Filter.class,
+    property = {
+        KeycloakOIDCFilter.CONFIG_FILE_PARAM + "=" + "keycloak.json",
+        HttpWhiteboardConstants.HTTP_WHITEBOARD_FILTER_PATTERN + "=" +"/*",
+        HttpWhiteboardConstants.HTTP_WHITEBOARD_CONTEXT_SELECT + "=" + "(osgi.http.whiteboard.context.name=mycontext)"
+    }
+)
+public class KeycloakFilter extends KeycloakOIDCFilter {
+  //
+}
+----
+
+To configure the component, either create a `mypackage.KeycloakFilter.cfg` file under the standard config location for your OSGi runtime:
+[source]
+
+----
+keycloak.config.file = /path/to/keycloak.json
+osgi.http.whiteboard.filter.pattern = /secure/*
+----
+
+or use interactive console, if your runtime allows for that:
+
+[source]
+----
+karaf@root()> config:edit mypackage.KeycloakFilter
+karaf@root()> config:property-set keycloak.config.file '${karaf.etc}/keycloak.json'
+karaf@root()> config:update
+----
+
+If you need more control, like e.g. providing custom `KeycloakConfigResolver` to implement <<_multi_tenancy,multi tenancy>>, you can register the filter programmatically:
+
+[source,java]
+----
+public class Activator implements BundleActivator {
+
+  private ServiceRegistration registration;
+
+  public void start(BundleContext context) throws Exception {
+    Hashtable props = new Hashtable();
+    props.put(HttpWhiteboardConstants.HTTP_WHITEBOARD_FILTER_PATTERN, "/secure/*");
+    props.put(KeycloakOIDCFilter.CONFIG_RESOLVER_PARAM, new MyConfigResolver());
+
+    this.registration = context.registerService(Filter.class.getName(), new KeycloakOIDCFilter(), props);
+  }
+
+  public void stop(BundleContext context) throws Exception {
+    this.registration.unregister();
+  }
+}
+----
+
+Please refer to http://felix.apache.org/documentation/subprojects/apache-felix-http-service.html#using-the-osgi-http-whiteboard[Apache Felix HTTP Service] for more info on programmatic registration.
```

Comment:
Another popular HTTP Service implementation is https://ops4j1.jira.com/wiki/spaces/paxweb/overview[OPS4j Pax-WEB] . If you are using Karaf, most likely you are using pax-web.